src/manifest.h

```diff
@@ -0,0 +1,38 @@
+#ifndef __MANIFEST__
+#define __MANIFEST__
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+struct manifest {
+	// Header
+	int manifest_version;
+	int version;
+	uint64_t filecount;
+	uint64_t contentsize;
+	struct list *includes; /* list of strings with manifest names */
+	//struct list *actions;     /* list of strings with actions */
+	char *component;
+
+	// File list
+	struct list *files;     /* struct file for files */
+	struct list *manifests; /* struct file for possible manifests */
+
+	// Helper data
+	struct list *submanifests; /* struct manifest for subscribed manifests */
+	unsigned int is_mix : 1;
+};
+
+struct manifest *manifest_parse(const char *component, const char *filename, bool header_only);
```

Comment:
Most other header files include a brief description of what the function is for, this one doesn't.

Reply:
haha, I totally forgot to add comments here.